### PR TITLE
fix(tooltip): Tooltip for disabled input does not disappear #3354

### PR DESCRIPTION
**Pull request by ddomingues**
Fixes the issue #3354

### CODE REVIEW

**Comment by chrisirhc** [comment]
Looks good. However, we're trying to avoid adding watchers (when the tooltip isn't displayed) as much as possible as there can be many tooltips on a page.

How about adding this watcher only when the tooltip is shown and removing it when it's hidden?

**Comment by ddomingues** [comment]
For me, looks good @chrisirhc.

**Comment by chrisirhc** [comment]
Nice, looks good. Just needs to be rebased.

**Comment by ddomingues** [comment]
@chrisirhc branch rebased, but when I did it, I observed that was did a commit similar to the issue #3354. Take a look https://github.com/ddomingues/bootstrap/commit/f5503279157893bd31dd52e989c569bb7b6f9e44#diff-e8665eb306ddd26399e5b80f00dae5bbR277.
However, the code doesn't evaluate the expression in a ng-disabled instead, it evaluates an attribute "disabled". I think it forces us to use this attribute in a model. What do you think?

File: src/tooltip/tooltip.js [hunk 1]
```diff
@@ -276,6 +284,14 @@ angular.module( 'ui.bootstrap.tooltip', [ 'ui.bootstrap.position', 'ui.bootstrap
               ttScope.title = val;
             });
 
+            function watchNgDisabled() {
+              unbindWatcher = scope.$watch(attrs.ngDisabled, function (newValue, oldValue) {
```
Comment (wesleycho) [inline src/tooltip/tooltip.js]: Is there any reason why you cannot do `attrs.$observe('disabled', ...)` here or something similar?

Comment (chrisirhc) [inline src/tooltip/tooltip.js]: I think that would work too. Though I haven't tested what $observe does for boolean attributes like this.

Comment (chrisirhc) [inline src/tooltip/tooltip.js]: I stand corrected, `$observe` is a much better idea, more performant (less watchers on scope) and direct to the point.

Thanks for mentioning this @wesleycho , I finally went to dig up how attribute observers work. They're only called on `$set` so it's great. :smile:

**Comment by ddomingues** [comment]
I think so. $observe is a much better solution. So I guest this pull request can be closed @chrisirhc, because it was implemented in an previous commit. Thanks for to mention @wesleycho :wink:

**Pull request closed without merging**